Compiler warnings

```
core/src/InfiniteDimensionalMCMCSampler.C: In member function 'void QUESO::InfiniteDimensionalMCMCSampler::_append_scalar_dataset(hid_t, double)':
core/src/InfiniteDimensionalMCMCSampler.C:279:9: warning: variable 'err' set but not used [-Wunused-but-set-variable]
core/src/InfiniteDimensionalMCMCSampler.C: In member function 'hid_t QUESO::InfiniteDimensionalMCMCSampler::_create_scalar_dataset(const string&)':
core/src/InfiniteDimensionalMCMCSampler.C:271:9: warning: 'dummy' may be used uninitialized in this function [-Wmaybe-uninitialized]
```

```
In file included from /opt/apps/ossw/libraries/boost/boost-1.51.0/sl6/gcc-4.7/include/boost/random/detail/large_arithmetic.hpp:19:0,
                 from /opt/apps/ossw/libraries/boost/boost-1.51.0/sl6/gcc-4.7/include/boost/random/detail/const_mod.hpp:23,
                 from /opt/apps/ossw/libraries/boost/boost-1.51.0/sl6/gcc-4.7/include/boost/random/linear_congruential.hpp:30,
                 from /opt/apps/ossw/libraries/boost/boost-1.51.0/sl6/gcc-4.7/include/boost/random/additive_combine.hpp:27,
                 from /opt/apps/ossw/libraries/boost/boost-1.51.0/sl6/gcc-4.7/include/boost/random.hpp:36,
                 from ../inc/queso/RngBoost.h:29,
                 from core/src/Environment.C:29:
/opt/apps/ossw/libraries/boost/boost-1.51.0/sl6/gcc-4.7/include/boost/random/detail/integer_log2.hpp:71:35: warning: always_inline function might not be inlinable [-Wattributes]
```

```
stats/src/TKGroup.C: In instantiation of 'QUESO::BaseTKGroup<V, M>::BaseTKGroup() [with V = QUESO::GslVector; M = QUESO::GslMatrix]':
stats/src/TKGroup.C:132:23:   required from here
stats/src/TKGroup.C:41:28: warning: passing NULL to non-pointer argument 1 of 'std::vector<_Tp, _Alloc>::vector(std::vector<_Tp, _Alloc>::size_type, const value_type&, const allocator_type&) [with _Tp = const QUESO::GslVector*; _Alloc = 
std::allocator<const QUESO::GslVector*>; std::vector<_Tp, _Alloc>::size_type = long unsigned int; std::vector<_Tp, _Alloc>::value_type = const QUESO::GslVector*; std::vector<_Tp, _Alloc>::allocator_type = std::allocator<const QUESO::GslV
ector*>]' [-Wconversion-null]
```

```
surrogates/src/InterpolationSurrogateBuilder.C: In instantiation of 'void QUESO::InterpolationSurrogateBuilder<V, M>::partition_work() [with V = QUESO::GslVector; M = QUESO::GslMatrix]':
surrogates/src/InterpolationSurrogateBuilder.C:233:23:   required from here
surrogates/src/InterpolationSurrogateBuilder.C:101:5: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
```

```
surrogates/src/InterpolationSurrogateIOASCII.C: In instantiation of 'void QUESO::InterpolationSurrogateIOASCII<V, M>::read(const string&, const QUESO::FullEnvironment&, const string&) [with V = QUESO::GslVector; M = QUESO::GslMatrix; std
::string = std::basic_string<char>]':
surrogates/src/InterpolationSurrogateIOASCII.C:252:23:   required from here
surrogates/src/InterpolationSurrogateIOASCII.C:60:5: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
surrogates/src/InterpolationSurrogateIOASCII.C:85:5: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
surrogates/src/InterpolationSurrogateIOASCII.C:109:5: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
surrogates/src/InterpolationSurrogateIOASCII.C:155:5: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
```

There are probably others, but this is what I found while trying to implement something.

--- a/examples/gp/pseudovector/gpmsa_pseudovector.C
+++ b/examples/gp/pseudovector/gpmsa_pseudovector.C
@@ -31,9 +31,6 @@ readData(const std::vector<QUESO::GslVector *> & simulationScenarios,
   double meansim = 0;
   double m2sim = 0;
 
-  double meanexpsim = 0;
-  double m2expsim = 0;
-
   // First line is a header, so we ignore it
   fgets(line, size, fp_in);
 

--- a/src/basic/src/ScalarSequence.C
+++ b/src/basic/src/ScalarSequence.C
@@ -1366,21 +1366,10 @@ ScalarSequence<T>::autoCorrViaFft(
   unsigned int    maxLag,
   std::vector<T>& autoCorrs) const
 {
-  unsigned int fftSize = 0;
-  {
-#warning WTF are 4 lines of unused code doing here? - RHS
-    double tmp = log((double) maxLag)/log(2.);
-    double fractionalPart = tmp - ((double) ((unsigned int) tmp));
-    if (fractionalPart > 0.) tmp += (1. - fractionalPart);
-    unsigned int fftSize1 = (unsigned int) std::pow(2.,tmp+1.); // Yes, tmp+1
-
-    tmp = log((double) numPos)/log(2.);
-    fractionalPart = tmp - ((double) ((unsigned int) tmp));
-    if (fractionalPart > 0.) tmp += (1. - fractionalPart);
-    unsigned int fftSize2 = (unsigned int) std::pow(2.,tmp+1);
-
-    fftSize = fftSize2;
-  }
+  double tmp = log((double) numPos)/log(2.);
+  double fractionalPart = tmp - ((double) ((unsigned int) tmp));
+  if (fractionalPart > 0.) tmp += (1. - fractionalPart);
+  unsigned int fftSize = (unsigned int) std::pow(2.,tmp+1);
 
   std::vector<double> rawDataVec(numPos,0.);
   std::vector<std::complex<double> > resultData(0,std::complex<double>(0.,0.));

--- a/src/gp/inc/GPMSA.h
+++ b/src/gp/inc/GPMSA.h
@@ -100,15 +100,6 @@ public:
   const std::vector<M> & m_observationErrorMatrices;
 
   //
-  // Intermediate calculations cached by factory
-  //
-  const V & residual;
-
-  const M & BT_Wy_B_inv;
-
-  const M & KT_K_inv;
-
-  //
   // Intermediate calculations we can cache
   //
   unsigned int num_svd_terms;
@@ -117,6 +108,15 @@ public:
   const M & m_experimentErrors;
 
   const ConcatenatedVectorRV<V, M> & m_totalPrior;
+
+  //
+  // Intermediate calculations cached by factory
+  //
+  const V & residual;
+
+  const M & BT_Wy_B_inv;
+
+  const M & KT_K_inv;
 };
 
 template <class V = GslVector, class M = GslMatrix>

--- a/src/surrogates/src/InterpolationSurrogateBuilder.C
+++ b/src/surrogates/src/InterpolationSurrogateBuilder.C
@@ -76,7 +76,7 @@ namespace QUESO
       }
 
     // Sanity check
-    queso_assert_equal_to( n_values, std::accumulate( m_njobs.begin(), m_njobs.end(), 0 ) );
+    queso_assert_equal_to( (int)n_values, std::accumulate( m_njobs.begin(), m_njobs.end(), 0 ) );
   }
 
   template<class V, class M>

--- a/src/surrogates/src/InterpolationSurrogateIOASCII.C
+++ b/src/surrogates/src/InterpolationSurrogateIOASCII.C
@@ -49,7 +49,7 @@ namespace QUESO
                                                  int reading_rank )
   {
     // Root processor
-    unsigned int root = reading_rank;
+    int root = reading_rank;
 
     MpiComm full_comm = env.fullComm();
 

--- a/test/test_pdfs/test_beta_pdf_gradient.C
+++ b/test/test_pdfs/test_beta_pdf_gradient.C
@@ -66,8 +66,8 @@ int main(int argc, char ** argv)
 
   // Here we test the gradient of the log pdf and the gradient of the pdf are
   // zero, as we expect.
-  double lnPdfValue1 = pdf.lnValue(point, NULL, &lnGradVector, NULL, NULL);
-  double pdfValue1 = pdf.actualValue(point, NULL, &gradVector, NULL, NULL);
+  pdf.lnValue(point, NULL, &lnGradVector, NULL, NULL);
+  pdf.actualValue(point, NULL, &gradVector, NULL, NULL);
 
   queso_require_less_equal_msg(std::abs(lnGradVector[0]), TOL, "grad log beta pdf values are incorrect");
   queso_require_less_equal_msg(std::abs(gradVector[0]), TOL, "grad beta pdf values are incorrect");

--- a/test/test_pdfs/test_gaussian_pdf_gradient.C
+++ b/test/test_pdfs/test_gaussian_pdf_gradient.C
@@ -71,13 +71,13 @@ int main(int argc, char ** argv)
   QUESO::GslVector point(mean);
 
   // We are testing that the gradient of log of the pdf is all zeros
-  double lnPdfValue1 = pdf.lnValue(point, NULL, &lnGradVector, NULL, NULL);
+  pdf.lnValue(point, NULL, &lnGradVector, NULL, NULL);
   queso_require_less_equal_msg(std::abs(lnGradVector[0]), TOL, "grad log gaussian pdf values are incorrect");
   queso_require_less_equal_msg(std::abs(lnGradVector[1]), TOL, "grad log gaussian pdf values are incorrect");
   queso_require_less_equal_msg(std::abs(lnGradVector[2]), TOL, "grad log gaussian pdf values are incorrect");
 
   // We are testing that the of the pdf is all zeros
-  double pdfValue1 = pdf.actualValue(point, NULL, &gradVector, NULL, NULL);
+  pdf.actualValue(point, NULL, &gradVector, NULL, NULL);
   queso_require_less_equal_msg(std::abs(gradVector[0]), TOL, "grad guassian pdf values are incorrect");
   queso_require_less_equal_msg(std::abs(gradVector[1]), TOL, "grad guassian pdf values are incorrect");
   queso_require_less_equal_msg(std::abs(gradVector[2]), TOL, "grad guassian pdf values are incorrect");
@@ -106,7 +106,7 @@ int main(int argc, char ** argv)
 
   QUESO::GaussianJointPdf<> pdf2("", paramDomain, mean, var);
 
-  lnPdfValue1 = pdf2.lnValue(point, NULL, &lnGradVector, NULL, NULL);
+  pdf2.lnValue(point, NULL, &lnGradVector, NULL, NULL);
 
   queso_require_less_equal_msg(std::abs(lnGradVector[0] - 1.0), TOL, "grad log gaussian pdf2 values are incorrect");
   queso_require_less_equal_msg(std::abs(lnGradVector[1] - 1.0), TOL, "grad log gaussian pdf2 values are incorrect");

--- a/test/test_pdfs/test_log_normal_pdf_gradient.C
+++ b/test/test_pdfs/test_log_normal_pdf_gradient.C
@@ -71,12 +71,12 @@ int main(int argc, char ** argv)
 
   // Here we test the gradient of the log of the pdf and the gradient of the
   // pdf are zero at the mode, as we expect.
-  double lnPdfValue1 = pdf.lnValue(point, NULL, &lnGradVector, NULL, NULL);
+  pdf.lnValue(point, NULL, &lnGradVector, NULL, NULL);
   queso_require_less_equal_msg(std::abs(lnGradVector[0]), TOL, "grad log log_normal pdf values are incorrect");
   queso_require_less_equal_msg(std::abs(lnGradVector[1]), TOL, "grad log log_normal pdf values are incorrect");
   queso_require_less_equal_msg(std::abs(lnGradVector[2]), TOL, "grad log log_normal pdf values are incorrect");
 
-  double pdfValue1 = pdf.actualValue(point, NULL, &gradVector, NULL, NULL);
+  pdf.actualValue(point, NULL, &gradVector, NULL, NULL);
   queso_require_less_equal_msg(std::abs(gradVector[0]), TOL, "grad log_normal pdf values are incorrect");
   queso_require_less_equal_msg(std::abs(gradVector[1]), TOL, "grad log_normal pdf values are incorrect");
   queso_require_less_equal_msg(std::abs(gradVector[2]), TOL, "grad log_normal pdf values are incorrect");

